screed/tests/test_attriberror.py

```diff
@@ -0,0 +1,87 @@
+import screed
+from screed.DBConstants import fileExtension
+import os
+
+
+class nostring:
+    def __init__(self):
+        self.exists = True
+
+    def __str__(self):
+        raise AttributeError
+
+    def __repr__(self):
+        raise AttributeError
+
+
+class test_comparisons():
+
+    def setup(self):
+        self._testfile = os.path.join(os.path.dirname(__file__), 'test.fa')
+        screed.read_fasta_sequences(self._testfile)
+
+        self._db = screed.ScreedDB(self._testfile)
+        self._ns = nostring()
+
+    def test_eq(self):
+        for k in self._db:
+            record = self._db.get(k)
+            try:
+                res = (record.sequence == self._ns)
+            except TypeError:
+                assert True
+            except:
+                assert False, "should have caught the TypeError"
```

Comment:
Move this up after line 30 if the exception is expected; otherwise this line is only evaluated when some other non-TypeError exception is thrown.